src/export/ExportFilePanel.cpp

```diff
@@ -371,23 +372,36 @@ void ExportFilePanel::ValidateAndFixExt()
 
    wxFileName filename;
    filename.SetFullName(mFullName->GetValue());
+   const auto desiredExt = filename.GetExt().Trim();
```

Comment:
So it's this `GetExt()` which is too lenient, right ? for `Artist – Track (feat. Artist)` returning ` Artist)` rather than an empty string.

The differentiation between the use cases where there is an extension but it doesn't match the format and where there is no valid extension isn't clear to me. I personally would have begun with building an extension getter function that behaves the proper way. But sure, it's hard to verify that this would really simplify the code below without trying it.